### PR TITLE
Carrion spider menu QOL

### PR DESCRIPTION
**Pull request by Alliostra**
## About The Pull Request

You can now assign carrion spiders to one or multiple groups, and activate all the spiders in a group at the same time.

The observer spider will also no longer be triggered when activating all spiders, since you don't really want that to happen if you hit "activate all".

Also fixes a bug with fuel tanks, you can now make them leak fuel and secure them to the floor again.

Hyperioo, Handyman and Shown were a great help as per usual.

## Why It's Good For The Game

makes it easier to pull off 9000 IQ schemes
carrionmenu

## Changelog
:cl:
add: Carrion spiders can now be assigned to groups to make them activateable at the same time.
tweak: the observer spider will no longer be triggered when activating all carrion spiders
fix: you can now secure water and fuel tanks to the ground and make fuel tanks leak fuel again
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/implant/implants/carrion/carrion_spiders.dm [hunk 1]
```diff
@@ -98,3 +103,12 @@
 
 /obj/item/implant/carrion_spider/proc/update_owner_mob()
 	owner_mob = owner_core.owner
+
+/obj/item/implant/carrion_spider/proc/toggle_group(group)
+	if(group == 1)
```
Comment (SirRichardFrancis) [inline code/game/objects/items/weapons/implant/implants/carrion/carrion_spiders.dm:108]: Extra checks don't really affect anything, but consider doing `switch` or `else if`'s

File: code/modules/organs/internal/carrion.dm [hunk 2]
```diff
@@ -141,9 +144,42 @@
 	if(href_list["activate_all"])
 		for(var/spider in active_spiders)
 			var/obj/item/implant/carrion_spider/CS = spider
-			if(istype(CS))
+			if(istype(CS) && CS.ignore_activate_all == FALSE)
```
Comment (SirRichardFrancis) [inline code/modules/organs/internal/carrion.dm:147]:
```suggestion
			if(istype(CS) && !CS.ignore_activate_all)
```

File: code/modules/organs/internal/carrion.dm [hunk 3]
```diff
@@ -141,9 +144,42 @@
...
 				CS.activate()
 
+	if(href_list["activate_group_1"])
+		for(var/spider in active_spiders)
+			var/obj/item/implant/carrion_spider/CS = spider
+			if(istype(CS) && CS.assigned_group_1 == TRUE)
```
Comment (SirRichardFrancis) [inline code/modules/organs/internal/carrion.dm:153]:
```suggestion
			if(istype(CS) && CS.assigned_group_1)
```

File: code/modules/organs/internal/carrion.dm [hunk 4]
```diff
@@ -141,9 +144,42 @@
...
+				CS.activate()
+
+	if(href_list["activate_group_2"])
+		for(var/spider in active_spiders)
+			var/obj/item/implant/carrion_spider/CS = spider
+			if(istype(CS) && CS.assigned_group_2 == TRUE)
```
Comment (SirRichardFrancis) [inline code/modules/organs/internal/carrion.dm:159]:
```suggestion
			if(istype(CS) && CS.assigned_group_2)
```

File: code/modules/organs/internal/carrion.dm [hunk 5]
```diff
@@ -141,9 +144,42 @@
...
+				CS.activate()
+
+	if(href_list["activate_group_3"])
+		for(var/spider in active_spiders)
+			var/obj/item/implant/carrion_spider/CS = spider
+			if(istype(CS) && CS.assigned_group_3 == TRUE)
```
Comment (SirRichardFrancis) [inline code/modules/organs/internal/carrion.dm:165]:
```suggestion
			if(istype(CS) && CS.assigned_group_3)
```

File: code/game/objects/items/weapons/implant/implants/carrion/carrion_spiders.dm [hunk 6]
```diff
@@ -98,3 +103,12 @@
...
+		assigned_group_1 = !assigned_group_1
+	if(group == 2)
+		assigned_group_2 = !assigned_group_2
+	if(group == 3)
+		assigned_group_3 = !assigned_group_3
```
Comment (SirRichardFrancis) [inline code/game/objects/items/weapons/implant/implants/carrion/carrion_spiders.dm:113]:
```suggestion
	if(group == 1)
		assigned_group_1 = !assigned_group_1
	else if(group == 2)
		assigned_group_2 = !assigned_group_2
	else if(group == 3)
		assigned_group_3 = !assigned_group_3	
```
```suggestion
	switch(group)
		if(1)
			assigned_group_1 = !assigned_group_1
		if(2)
			assigned_group_2 = !assigned_group_2
		if(3)
			assigned_group_3 = !assigned_group_3	
```

File: code/game/objects/items/weapons/implant/implants/carrion/carrion_spiders.dm (hunk 1 above)
Comment (Alliostra) [reply code/game/objects/items/weapons/implant/implants/carrion/carrion_spiders.dm]: you already did that for me, so. Gonna resolve this and thanks

**Review by TheShown911 — APPROVED** [review]
Good.

File: code/game/objects/items/weapons/implant/implants/carrion/carrion_spiders.dm [hunk 7]
```diff
@@ -11,6 +11,11 @@
 	var/gene_price = 0
 	var/do_gibs = TRUE
 	var/last_stun_time = 0 //Used to avoid cheese
+	var/ignore_activate_all = FALSE
+
+	var/assigned_group_1 = FALSE
+	var/assigned_group_2 = FALSE	
+	var/assigned_group_3 = FALSE
```
Comment (Shadow-Quill) [inline code/game/objects/items/weapons/implant/implants/carrion/carrion_spiders.dm:18]: You could make these into a bitflag, which means in the future groups can be added/removed easily.

File: code/modules/organs/internal/carrion.dm [hunk 8]
```diff
@@ -141,9 +144,42 @@
...
+
+	if(href_list["toggle_group_2"])
+		var/obj/item/implant/carrion_spider/activated_spider = locate(href_list["toggle_group_2"]) in active_spiders
+		if(activated_spider)
+			activated_spider.toggle_group(2)
+
+	if(href_list["toggle_group_3"])
+		var/obj/item/implant/carrion_spider/activated_spider = locate(href_list["toggle_group_3"]) in active_spiders
+		if(activated_spider)
+			activated_spider.toggle_group(3)
```
Comment (Shadow-Quill) [inline code/modules/organs/internal/carrion.dm:181]: Re: bitflags, doing so would mean these lines could be cut down to a block to activate spiders in group X, and a block to toggle group X on a specific spider.

File: nano/templates/carrion_spiders.tmpl [hunk 9]
```diff
@@ -8,17 +8,43 @@ Used In File(s): code\modules\organs\internal\carrion.dm
...
+		{{if value.assigned_group_2}}
+			<td>{{:helper.link('', 'check', {'toggle_group_2' : value.spider})}}</td>
+		{{else}}
+			<td>{{:helper.link('', 'close', {'toggle_group_2' : value.spider})}}</td>
+		{{/if}}
+		{{if value.assigned_group_3}}
+			<td>{{:helper.link('', 'check', {'toggle_group_3' : value.spider})}}</td>
+		{{else}}
+			<td>{{:helper.link('', 'close' , {'toggle_group_3' : value.spider})}}</td>
+		{{/if}}
```
Comment (Shadow-Quill) [inline nano/templates/carrion_spiders.tmpl:47]: You can get rid of the if/elses by using a ternary operator for the button icon.

File: code/game/objects/items/weapons/implant/implants/carrion/carrion_spiders.dm (hunk 7 above)
Comment (Alliostra) [reply code/game/objects/items/weapons/implant/implants/carrion/carrion_spiders.dm:18]: I'd have to look into how these work (I don't actually know how to code) and make these changes in a different PR so this one can hopefully get merged for this launch.

**Pull request merged**